frontend/dockerfile/dockerfile_test.go

```diff
@@ -6393,6 +6403,83 @@ ARG BUILDKIT_SBOM_SCAN_STAGE=true
 	require.Equal(t, 1, len(att.LayersRaw))
 }
 
+func testReproSourceDateEpoch(t *testing.T, sb integration.Sandbox) {
+	integration.CheckFeatureCompat(t, sb, integration.FeatureOCIExporter, integration.FeatureSourceDateEpoch)
+	if sb.Snapshotter() == "native" {
+		t.Skip("the digest is not reproducible with the \"native\" snapshotter because hardlinks are processed in a different way: https://github.com/moby/buildkit/pull/3456#discussion_r1062650263")
+	}
+	f := getFrontend(t, sb)
+
+	tm := time.Date(2015, time.October, 21, 7, 28, 0, 0, time.UTC)
+	t.Logf("SOURCE_DATE_EPOCH=%d", tm.Unix())
+
+	dockerfile := []byte(`# The base image cannot be busybox, due to https://github.com/moby/buildkit/issues/3455
+FROM --platform=linux/amd64 debian:bullseye-20221219-slim@sha256:171530d298096f0697da36b3324182e872db77c66452b85783ea893680cc1b62
```

Comment:
alpine doesn't work either? If possible would like to avoid another non-mirrored image.

Reply:
Alpine's `/bin/find` lacks `-newermt`, `-writable`, etc.

Reply:
if we can't find a better solution, we might still need to add it to mirrors. Otherwise I think even different workers and frontend types in build matrix will each do their own pull.

Reply:
> Alpine's /bin/find lacks -newermt, -writable, etc.

`apk add findutils` ?

Reply:
> > Alpine's /bin/find lacks -newermt, -writable, etc.
> 
> `apk add findutils` ?

That will result in an unreproducible build, and pinning the apk version is hard

Reply:
> mirror

Does `integration.WithMirroredImages` in L227 suffice?

Reply:
> That will result in an unreproducible build, and pinning the apk version is hard

Correct. https://github.com/tonistiigi/buildkit-alpine

> Does integration.WithMirroredImages in L227 suffice?

But that is not with a digest? Also pin it directly to amd64-only digest if the test is amd64 specific.

Reply:
Updated to amd64-only digest

Reply:
Doesn't pass in arm anymore 😢 

```
    run.go:158: 
                Error Trace:    /src/frontend/dockerfile/run.go:158
                                                        /src/frontend/dockerfile/dockerfile_test.go:228
                Error:          Received unexpected error:
                                docker.io/arm64v8/debian:bullseye-20230109-slim@sha256:1acb06a0c31fb467eb8327ad361f1091ab265e0bf26d452dea45dcb0c0ea5e75: not found
                Test:           TestIntegration
```

Reply:
If you put the pins in https://github.com/moby/buildkit/blob/master/util/testutil/integration/pins.go (and use amd64 digest for arm, with a comment) then I think it should work.

Reply:
updated